.pre-commit-hooks.yaml

```
@@ -1,7 +1,8 @@
 - id: sphinx-lint
-  name: Sphinx lint
+  name: Sphinx Lint
   description: 'Searches for common typos in sphinx-flavored rst files.'
   files: '\.rst$'
```

Would this also be a good idea?

```suggestion
  files: '\.rst$'
  types: [rst]
```

---

What's the difference between the two?

---

The two options work in different ways. `files` is a regex pattern that specifies exactly which file paths from the repo should be passed to the tool. `types` is a list of file types that should be passed to the tool. There's detailed docs on how pre-commit classifies files by type here: https://pre-commit.com/#filtering-files-with-types

People might want to override `files` in the `.pre-commit-config.yaml` file (because they only care about running Sphinx Lint on a specific directory), but still want to specify that Sphinx Lint should only run on `.rst` files. For example, see the CPython config:

https://github.com/python/cpython/blob/fa18b0afe47615dbda15407a102b84e40cadf6a5/.pre-commit-config.yaml#L31-L32

---

Yes, good idea.